MIP/mip-16/README.md

```diff
@@ -0,0 +1,193 @@
+
+# MIP-16: Calculation of gas fees 
+
+- **Description**: How to calculate gas fees for transactions on Movement L2.
+- **Authors**: [Franck Cassez](mailto:franck.cassez@movementlabs.xyz)
+
+
+## Abstract
+This MIP describes the structure of the gas fees and overall resource consumption needed to execute, and post a transaction to the DA/L1.
+
+
+## Motivation
+
+The transaction fees on L2s are multi-dimensional [[9](#ref-9), [10](#ref-10)]. They comprise of _execution fees_, _settlement fees_ and _data availability fees_.
+This MIP describes what the different types of fees are, and how the transaction fees can be computed.
+
+## Scope of this MIP
+This MIP applies to costs/fees **related to executing Move bytecode**. 
+
+> [!WARNING] 
+> It does not cover the extended future framework where EVM bytecode can be executed on a Movement chain. 
+
+Althought the costs (in terms of resources consumption) are expressed in gas units when executing Move and EVM bytecode, the gas units are **not comparable**, e.g. the minimum gas for an EVM transaction is 21000 gas units, whereas for Move-Aptos it is 700 gas units. As a result designing a fee mechanism for Movement chains that support Move and EVM bytecodes will require some careful adjustments and analysis of the different fee structures (Move and EVM). 
+
+> [!WARNING] 
+> This MIP does not cover _staking_ (and slashing and rewards) for validators.
+
+## Fee structure 
+
+The processing costs on rollups [[1](#ref-1) - [6](#ref-6)] are  [multi-dimensional](https://medium.com/offchainlabs/understanding-arbitrum-2-dimensional-fees-fd1d582596c9). The cost of processing a transaction on a Movement chain consists of:
+
+1. **execution** fees: the costs that correspond to resources usage, CPU and permanent storage.
+2. **data availability** fees: the costs of publishing transactions/blocks data to a _data availability layer_.
+3. **settlement** fees: the costs of _validating_ a transaction (e.g. zk-proof generation and verification, validators attestations.)
+
+Overall the transaction fees are defined by the sum of three types of fees:
+
+$$ {ExecFees} + {DAFees} + {SetlFees}\mathpunct. $$
+
+The fees should be expressed in a single (crypto)-currency, e.g. USD, or a token `$APT`, `$MOVE`, `$ETH`.
+
+### Execution 
+
+#### MoveVM execution fees 
+
+To execute a transaction and store the result, the operator of the chain uses some hardware: CPU and disk storge. The **execution** fees represent the compensation of the operator for executing a transaction. 
```

Comment:
```suggestion
To execute a transaction and store the result, the operator (and other entities, e.g. validator nodes) of the chain uses some hardware: CPU and disk storge. The **execution** fees represent the compensation of the operator (and e.g. validator nodes) for executing a transaction. 
```

Reply:
in our setting it is infact the validators that carry most of the burden, not the operator. The operator has costs to provide public good explorers etc (for which admittedly the operator also has to run a full node, but that is one full node compared to N validators). Similarly in a fraud proof system the same applies for watchtower nodes